src/Sylius/Bundle/CartBundle/Controller/CartItemController.php

```diff
@@ -53,49 +39,61 @@ public function addAction(Request $request)
     {
         $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
 
-        $cart = $this->getCurrentCart();
-        $emptyItem = $this->newResourceFactory->create($configuration, $this->factory);
+        $this->isGrantedOr403($configuration, ResourceActions::CREATE);
+        $newResource = $this->newResourceFactory->create($configuration, $this->factory);
 
-        $eventDispatcher = $this->getEventDispatcher();
+        $itemQuantityModifier = $this->get('sylius.order_item_quantity_modifier');
```

Comment:
Let's extract this to a private method (getQuantityModifier), just for now, later we consider further refactoring.